### PR TITLE
Fix branch-23.06 forward-merge into branch-23.08

### PR DESCRIPTION
**Pull request by trxcllnt**
Fixes merge conflicts in https://github.com/rapidsai/cuspatial/pull/1171.

### CODE REVIEW

**Review by isVoid — APPROVED** [review]

File: python/cuspatial/cuspatial/__init__.py [hunk 1]
```diff
@@ -28,5 +29,4 @@
 )
 from .io.geopandas import from_geopandas
 
-__version__ = get_versions()["version"]
-del get_versions
+__version__ = "23.06.00"
```
Comment (bdice) [inline python/cuspatial/cuspatial/__init__.py:32]: Please verify that `ci/release/update-version.sh` is correct.
```suggestion
__version__ = "23.08.00"
```

**Review by bdice — CHANGES_REQUESTED** [review]

File: python/cuspatial/cuspatial/__init__.py (hunk 1 above)
Comment (ajschmidt8) [reply python/cuspatial/cuspatial/__init__.py:32]: I'm thinking we might want to do those changes in another PR.

Typically we try to limit the scope of these PRs to just fix the conflicts.

Comment (ajschmidt8) [reply python/cuspatial/cuspatial/__init__.py:32]: Unless this change is part of the conflict

**Comment by bdice** [comment]
Changes are needed in GitHub Actions files, Python `__init__.py`:
```
.github/workflows/build.yaml:    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-build.yml@branch-23.06
.github/workflows/build.yaml:    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-publish.yml@branch-23.06
.github/workflows/pr.yaml:    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-build.yml@branch-23.06
.github/workflows/pr.yaml:    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-test.yml@branch-23.06
.github/workflows/test.yaml:    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-test.yml@branch-23.06
python/cuspatial/cuspatial/__init__.py:__version__ = "23.06.00"
```

Changes may be needed in devcontainers files (I'm not familiar with how these are maintained):
```
.devcontainer/Dockerfile:ARG RAPIDS=23.06
.devcontainer/pip/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/conda/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/pip/isolated/.devcontainer/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/pip/unified/.devcontainer/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/conda/unified/.devcontainer/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
.devcontainer/conda/isolated/.devcontainer/devcontainer.json:    "ghcr.io/rapidsai/devcontainers/features/rapids-build-utils:23.6": {}
```

File: python/cuspatial/cuspatial/__init__.py (hunk 1 above)
Comment (trxcllnt) [reply python/cuspatial/cuspatial/__init__.py]: `update-version.sh` is correct, I just assumed git chose the correct versions in the merge commit.

**Comment by bdice** [comment]
This change is needed: https://github.com/rapidsai/raft/pull/1556#issuecomment-1568971948

This is the diff you need: https://github.com/rapidsai/raft/pull/1556/files#diff-8a0188e09c831b0e6ca8c1c51ab640efd1e519b1d7f94ab6251c9f0b90a82304R64-R69

If you run `ci/release/update-version.sh 23.10.00` and then grep for `23.8`, you can see the line `- &rmm_conda rmm==23.8.*` in `dependencies.yaml` didn't change.

**Review by raydouglass — APPROVED** [review]

**Pull request merged**